sync_module/selective_sync/selective_sync.go

```diff
@@ -0,0 +1,161 @@
+// Package selectivesync implements a datasync manager.
+package selectivesync
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/datamanager"
+	"go.viam.com/rdk/services/vision"
+)
+
+// Model is the full model definition.
+var Model = resource.NewModel("selectivesync", "demo", "vision")
+
+func init() {
+	registration := resource.Registration[resource.Resource, *Config]{
+		Constructor: newSelectiveSyncer,
+	}
+	resource.RegisterComponent(generic.API, Model, registration)
+}
+
+// Config contains the name to the underlying camera and the name of the vision service to be used.
+type Config struct {
+	DataManager   string `json:"data_manager"`
+	Camera        string `json:"camera"`
+	VisionService string `json:"vision_service"`
+}
+
+// Validate validates the config and returns implicit dependencies.
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.DataManager == "" {
+		return nil, fmt.Errorf(`expected "data_manager" attribute in %q`, path)
+	}
+	if cfg.Camera == "" {
+		return nil, fmt.Errorf(`expected "camera" attribute in %q`, path)
+	}
+	if cfg.VisionService == "" {
+		return nil, fmt.Errorf(`expected "vision_service" attribute in %q`, path)
+	}
+
+	return []string{cfg.DataManager, cfg.Camera, cfg.VisionService}, nil
+}
+
+type visionSyncer struct {
+	resource.Named
+	actualDM      datamanager.Service
+	camera        camera.Camera
+	visionService vision.Service
+
+	cancelCtx  context.Context
+	cancelFunc func()
+	wg         sync.WaitGroup
+	logger     golog.Logger
+}
+
+func newSelectiveSyncer(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger,
+) (resource.Resource, error) {
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	v := &visionSyncer{
+		Named:      conf.ResourceName().AsNamed(),
+		logger:     logger,
+		cancelCtx:  cancelCtx,
+		cancelFunc: cancelFunc,
+	}
+	if err := v.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	v.wg.Add(1)
+	defer v.wg.Done()
```

Comment:
Ah sorry I meant removing the goroutine kick off within the `startBackgroundProcess` function. To guarantee that we don't have a resource leak, we need to make sure `v.wg.Done()` is not called until the background routine returns. Since `startBackgroundProcess` returns immediately, the wg will be decremented immediately, even though the background process is still running.  

This can be fixed by either:
- Making `v.startBackgroundProcess()` synchronous and kicking off the goroutine here (with a deferred `.Done()` call after it returns).
- Passing the wait group into `v.startBackgroundProcess()` and not calling Done until the inner goroutine there returns.

I think the former would be a bit more clear but both would work